packages/noya-state/src/selection.ts

```diff
@@ -41,3 +44,32 @@ export function getDragHandles(
     };
   });
 }
+
+export function getLineDragHandles(
+  boundingRect: Rect,
+  points: CurvePoint[],
+  handleSize: number = dragHandelSize,
+): DragHandle[] {
+  const startDecodedPoint = decodeCurvePoint(points[0], boundingRect);
+  const endDecodedPoint = decodeCurvePoint(points[1], boundingRect);
+
+  return compassDirections
+    .filter(
+      (compassDirection) =>
+        compassDirection === 'e' || compassDirection === 'w',
+    )
+    .map((compassDirection) => {
+      const directionPoint =
+        compassDirection === 'w' ? startDecodedPoint : endDecodedPoint;
+
+      return {
+        rect: {
+          x: directionPoint.point.x - handleSize / 2,
+          y: directionPoint.point.y - handleSize / 2,
+          width: handleSize,
+          height: handleSize,
+        },
+        compassDirection,
```

Comment:
If it's not too hard, we might want to pick the direction based on angle like Sketch. If it's hard, this is fine for now (I will have some other small comments if we keep this)

Reply:
I added some code to calculate the angle between start and end points, however I'm having a hard time finding the relationship between the angle and the direction.

Reply:
Also would we use this angle or the rotation on layer.rotation?

Reply:
I'm not sure, but let's get the drag handles rendering in the right positions, cause otherwise I think it'll be hard to tell if this is working or not